static/js/settings.js

```diff
@@ -55,14 +55,70 @@ const SettingsScreen = {
     this.availableAddons = new Map();
     this.installedAddons = new Map();
     this.fetchAddonDeferred = null;
+
+    this.insertTitleElement(this.menu, 'Settings',
+                            '/optimized-images/settings-icon.png');
+    this.insertTitleElement(this.domainSettings, 'Domain',
+                            '/optimized-images/domain-icon.png');
+    this.insertTitleElement(this.userSettingsMain, 'Users',
+                            '/optimized-images/users-icon.png');
+    this.insertTitleElement(this.userSettingsEdit, 'Edit User',
+                            '/optimized-images/user.svg');
+    this.insertTitleElement(this.userSettingsAdd, 'Add User',
+                            '/optimized-images/user.svg');
+    this.insertTitleElement(this.adapterSettings, 'Adapters',
+                            '/optimized-image/adapters-icon.png');
+    this.insertTitleElement(this.addonMainSettings, 'Add-ons',
+                            '/optimized-images/add-on.svg');
+    const addonConfigTitle =
+      this.insertTitleElement(this.addonConfigSettings,
+                              'Configure Add-on',
+                              '/optimized-images/add-on.svg');
+    this.addonConfigTitleName =
+      addonConfigTitle.querySelector('.section-title-name');
+
+    this.insertTitleElement(this.addonDiscoverySettings,
+                            'Discover New Add-ons',
+                            '/optimized-images/add-on.svg');
+    this.insertTitleElement(this.experimentSettings, 'Experiments',
+                            '/optimized-images/experiments-icon.png');
+    this.insertTitleElement(this.updateSettings, 'Updates',
+                            '/optimized-images/update-icon.svg');
+    this.insertTitleElement(this.authorizationSettings, 'Authorizations',
+                            '/optimized-images/authorization.svg');
+    this.insertTitleElement(this.developerSettings, 'Developer',
+                            '/optimized-images/developer-icon.svg');
+
+    this.discoverAddonsButton.addEventListener('click', () => {
+      page('/settings/addons/discovered');
+    });
+    this.addUserButton.addEventListener('click', () => {
+      page('/settings/users/add');
+    });
+  },
+
+  insertTitleElement: function(section, name, icon) {
+    const elt = document.createElement('div');
+    elt.classList.add('section-title');
+    elt.innerHTML = `
+      <div class="section-title-back-flex"></div>
+      <div class="section-title-container">
+        <img class="section-title-icon" alt="Section Title" src="${icon}" />
+        <span class="section-title-name">${name}</span>
+      </div>
+      <div class="section-title-speech-flex"></div>
```

Comment:
I realise this is the HTML that was already in use (I hadn't noticed before), but is it possible to centre align this title without using empty these empty divs? The flex approach is working brilliantly in some cases.

![screen shot 2018-08-24 at 12 33 13](https://user-images.githubusercontent.com/552417/44582558-e47c8200-a799-11e8-8717-e823f30a7123.png)

Ideally this would be a more semantically meaningful HTML tag like h2, centered with text-align or margin: auto. Perhaps with a width of 100% and padding to account for the icons.

Reply:
I took the flex approach originally after considering all other easier alternatives. The key thing here is that at significantly limited screen widths (like 240px) the icon can overlap with the back button. There's an unrelated issue here which is making the approach look less clever which is that settings-section has a 1.5rem padding not taken into account by the title.

If we do want to modify this approach in the future, a change to insertTitleElement is just a couple of keystrokes away. The complexity of the current solution is why I avoided hard-coding all the HTML for the title element as it would make further reconsideration of the design difficult.